Sources/MapboxDirectionsCLI/ProcessCommand.swift

```diff
@@ -53,6 +56,22 @@ class ProcessCommand<ResponceType : Codable, OptionsType : DirectionsOptions > :
                let jsonData = try? JSONSerialization.data(withJSONObject: object, options: [.prettyPrinted]) {
                 outputText = String(data: jsonData, encoding: .utf8)!
             }
+        case .gpx:
+            var gpxText: String = String("<?xml version=\"1.0\" encoding=\"UTF-8\"?>")
+            gpxText.append("\n<gpx")
+            // do we need to include additional tags like metadata or the schema version?
+            
+            guard let routeResponse = routeResponse else { return }
+            guard let routes = routeResponse.routes else { return }
+            routes.forEach { route in
+                let shape = route.shape
+                for coord in shape!.coordinates {
+                    gpxText.append("\n<wpt lat=\"\(coord.latitude)\" lon=\"\(coord.longitude)\">")
+                    gpxText.append("</wpt>")
+                }
```

Comment:
- Do you know how far away each coordinate from each other? 
- What about timestamp of the coordinate? 
- Have you tried running this gpx in Xcode?

Reply:
I meant to open this as a draft. I am working on adding the timestamps now.

Reply:
Timestamps are added (using `total distance/expected time`, and the output gpx works with Xcode. I'll check with a multi-leg trip too.